src/Data/Text.hs

```diff
@@ -1415,7 +1325,7 @@ drop :: Int -> Text -> Text
 drop n t@(Text arr off len)
     | n <= 0    = t
     | n >= len  = empty
-    | otherwise = if m >= 0 then text arr (off+m) (len-m) else mempty
+    | otherwise = if m >= 0 then Text arr (off+m) (len-m) else mempty
```

Comment:
Good point. I assumed the `n` check would be catching this but `n` is not the same as `m`. I think I'll give this family of functions a second look. I think we can simplify this

Reply:
Let me know what you think of f0d0f6f1983bfd6e8200b1a32e8fbbbe16de3f0c . I've put in more checks against `m` rather than `n`. This means we do fewer checks but I guess we have to calculate `m` more often